base/samples/Transformation.hpp

```diff
@@ -0,0 +1,90 @@
+#ifndef BASE_SAMPLES_TRANSFORMATION_HPP
+#define BASE_SAMPLES_TRANSFORMATION_HPP
+
+#include <string>
+#include <base/Time.hpp>
+#include <base/TransformWithCovariance.hpp>
+#include <base/samples/RigidBodyState.hpp>
+
+namespace base { namespace samples {
+
+/**
+ * Represents a transformation in 3D of a local coordinate frame inside a global coordinate frame.
+ * A frame ID shall be globally unique and corresponds to a coordinate system.
+ * Following the [Rock's conventions](http://rock.opendfki.de/wiki/WikiStart/Standards)
+ * coordinate systems are right handed with X forward, Y left and Z up.
+ *
+ * The transformation \f$ ^{global}_{local}T \f$ in this structure transforms a vector \f$ v_{local} \f$
+ * from the local into the global frame:
+ * \f$ v_{global} = ^{global}_{local}T \cdot v_{local} \f$
+ * In other words the transformation is the pose of the local frame expressed in the global frame.
+ * Example of a transformation chain:
+ * \f$ ^{world}_{sensor}T = ^{world}_{body}T \cdot ^{body}_{sensor}T \f$
+ */
+class Transformation
+{
+public:
+    /** Reference timestamp of the transformation sample */
+    base::Time time;
+
+    /** ID of the global reference frame */
+    std::string global_frame_id;
```

Comment:
The motivation for the naming was that "global" is "more global", but I agree that the naming is not perfect. I don't think ``reference_frame_id`` will be entirely clear either (I assume you mean the "more global" frame).
How about ``base_frame`` and ``local_frame`` or ``parent_frame`` and ``child_frame``?
O.t.o.h., if it is mostly a documentation thing, we could stick with the RBS nomenclature but make the documentation more clear.

We should actually also make it more clear (and more consistent) in what frame the covariance information is stored.

Besides the nomenclature, another intention of ``Transformation`` is to have a type which represents the full covariance (with cross-covariances) of 3D transformations while not having to store velocity information (which often is not needed/not available).

Regarding the docu of RBS: I always assumed that it expresses the Pose of the "source" inside the "target", i.e., when extracting an ``Eigen::Transform T`` of the RBS and I have a vector ``x_source`` in the source frame, I can calculate ``x_target=T*x_source``. (At least that is how ``operator Eigen::Transform<...>`` is implemented).
The documentation however says that the orientation is the orientation of the target frame inside the source frame, whereas the position is the position of the source frame inside the target frame (only the latter is consistent with the implementation).
For velocities it is inconsistent again: translational velocities are expressed in the target frame, whereas angular velocities are expressed in the "body fixed frame" (which I assume means the source frame).
But as there are no interactions between velocities and the Pose (inside the implementation of RBS), this is likely handled differently by different users of RBS ...